stdlib/3/xmlrpc/client.pyi

```diff
@@ -0,0 +1,279 @@
+import io
+import sys
+import time
+import gzip
+import http.client
+
+from typing import Any, Callable, Dict, IO, Iterable, List, Mapping, Optional, Protocol, Text, Tuple, Type, TypeVar, Union, overload
+from types import TracebackType
+from datetime import datetime
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+_T = TypeVar("_T")
+class _HasTimeTuple(Protocol):
+    def timetuple(self) -> time.struct_time: ...
+_DateTimeComparable = Union[DateTime, datetime, str, _HasTimeTuple]
+_Marshallable = Union[None, bool, int, float, str, bytes, tuple, list, dict, datetime, DateTime, Binary]
+_XMLDate = Union[int, datetime, Tuple[int, ...], time.struct_time]
+_HostType = Union[Tuple[str, Dict[str, str]], str]
+
+def escape(s: str) -> str: ...  # undocumented
+
+PARSE_ERROR: int  # undocumented
+SERVER_ERROR: int  # undocumented
+APPLICATION_ERROR: int  # undocumented
+SYSTEM_ERROR: int  # undocumented
+TRANSPORT_ERROR: int  # undocumented
+
+NOT_WELLFORMED_ERROR: int  # undocumented
+UNSUPPORTED_ENCODING: int  # undocumented
+INVALID_ENCODING_CHAR: int  # undocumented
+INVALID_XMLRPC: int  # undocumented
+METHOD_NOT_FOUND: int  # undocumented
+INVALID_METHOD_PARAMS: int  # undocumented
+INTERNAL_ERROR: int  # undocumented
+
+class Error(Exception): ...
+
+class ProtocolError(Error):
+
+    url: str
+    errcode: int
+    errmsg: str
+    headers: Dict[str, str]
+
+    def __init__(self, url: str, errcode: int, errmsg: str, headers: Dict[str, str]) -> None: ...
+
+class ResponseError(Error): ...
+
+class Fault(Error):
+
+    faultCode: str
+    faultString: str
+
+    def __init__(self, faultCode: str, faultString: str, **extra: Any) -> None: ...
+
+
+boolean = bool
+Boolean = bool
+
+def _iso8601_format(value: datetime) -> str: ...  # undocumented
+def _strftime(value: _XMLDate) -> str: ...  # undocumented
+
+class DateTime:
+
+    value: str  # undocumented
+
+    def __init__(self, value: Union[int, str, datetime, time.struct_time, Tuple[int, ...]] = ...): ...
+    def __lt__(self, other: _DateTimeComparable) -> bool: ...
+    def __le__(self, other: _DateTimeComparable) -> bool: ...
+    def __gt__(self, other: _DateTimeComparable) -> bool: ...
+    def __ge__(self, other: _DateTimeComparable) -> bool: ...
+    def __eq__(self, other: _DateTimeComparable) -> bool: ...  # type: ignore
+    def make_comparable(self, other: _DateTimeComparable) -> Tuple[str, str]: ...  # undocumented
+    def timetuple(self) -> time.struct_time: ...  # undocumented
+    def decode(self, data: Any) -> None: ...
+    def encode(self, out: IO[str]) -> None: ...
```

Comment:
Instead of `IO`, prefer using an ad-hoc protocol with just the methods you need (in this case, just `def write(self, __o: str) -> None:`).